data/helpers.d/setting

```diff
@@ -158,48 +158,155 @@ ynh_webpath_register () {
 #    re:domain.tld/app/api/[A-Z]*$ -> domain.tld/app/api/[A-Z]*$
 #
+# Note that globally the parameter 'url' and 'additional_urls' are same. The only difference is:
+# - 'url' is only one url, 'additional_urls' can be a list of urls. There are no limitation of 'additional_urls'
+# - 'url' is used for the url of tile in the SSO (if enabled with the 'show_tile' parameter)
+#
+#
+# About the authentication header (auth_header parameter).
+# The SSO pass (by default) to the application theses following HTTP header (linked to the authenticated user) to the application:
+#        - "Auth-User": username
+#        - "Remote-User": username
+#        - "Email": user email
+#
+# Generally this feature is usefull to authenticate automatically the user in the application but in some case the application don't work with theses header and theses header need to be disabled to have the application to work correctly. 
+# See https://github.com/YunoHost/issues/issues/1420 for more informations
+#
+#
 # Requires YunoHost version 3.7.0 or higher.
 ynh_permission_create() {
     # Declare an array to define the options of this helper.
-    local legacy_args=pua
-    local -A args_array=( [p]=permission= [u]=url= [a]=allowed= )
+    local legacy_args=puAhaltP
+    local -A args_array=( [p]=permission= [u]=url= [A]=additional_urls= [h]=auth_header= [a]=allowed= [l]=label= [t]=show_tile= [P]=protected= )
     local permission
     local url
+    local additional_urls
+    local auth_header
     local allowed
+    local label
+    local show_tile
+    local protected
     ynh_handle_getopts_args "$@"
     url=${url:-}
+    additional_urls=${additional_urls:-}
+    auth_header=${auth_header:-}
     allowed=${allowed:-}
+    label=${label:-}
+    show_tile=${show_tile:-}
+    protected=${protected:-}
 
     if [[ -n $url ]]
     then
-        url="'$url'"
+        url=",url='$url'"
+    fi
+
+    if [[ -n $additional_urls ]]
+    then
+        # Convert a list from getopts to python list
+        # Note that getopts separate the args with ';'
+        # By example:
+        # --additional_urls /urlA /urlB
+        # will be:
+        # additional_urls=['/urlA', '/urlB']
+        additional_urls=",additional_urls=['${additional_urls//;/\',\'}']"
+    fi
+
+    if [[ -n $auth_header ]]
+    then
+        if [ $auth_header == "true" ]
+        then
+            auth_header=",auth_header=True"
+        else
+            auth_header=",auth_header=False"
+        fi
+    fi
+
+    if [[ -n $allowed ]]
+    then
+        # Convert a list from getopts to python list
+        # Note that getopts separate the args with ';'
+        # By example:
+        # --allowed alice bob
+        # will be:
+        # allowed=['alice', 'bob']
+        allowed=",allowed=['${allowed//;/\',\'}']"
+    fi
+
+    if [[ -n ${label:-} ]]; then
+        label=",label='$label'"
     else
-        url="None"
+        label=",label='$YNH_APP_LABEL ($permission)'"
     fi
```

Comment:
Do we foresee ways for people to edit "sub-permissions" labels ?

Otherwise it sounds to me like the current implementation means that the label of a subpermission is pretty much permanent (no mean for the admin to change it) which may be confusing if : 

- main app label is e.g. `Wordpress`
- there's a sub-permission for the admin, e.g. `Wordpress (admin)`
- the admin changes the label of the app to `My Blog`
- .. but the wordpress's admin tile remains named "Wordpress (admin)`
- now imagine if you have 3+ wordpress installed with 3 tiles all named "Wordpress (admin)" and no way to change it ? :s 

So imho, when creating a subpermission, we should only define the label of that permission, but in the end the displayed text should always be reconstructed as `<the app label> (<the subperm label>)` ?

Reply:
We can do this when we rebuild the SSOwat configuration.